<html lang="en">

<head>
    <meta charset="utf-8">
    <link rel="stylesheet" href="/static/style/base.css" type="text/css">
    <style type="text/css">
        body {
            font-size: 16px;
        }

        .padding-m {
            padding: 20px 0px 20px 0px;

        }

        .mainbody {
            padding: 20px 0px 35px;
        }

        .footer {
            background-color: #f8f9fa;
        }

        figcaption {
            padding-bottom: 12px;
            padding-top: 6px;
            max-width: 90%;
            margin-bottom: 20px;
            font-style: italic;
            color: gray;
        }

        .docimage {
            margin: 20px 0px 20px 0px;
        }

        h2 {
            font-size: 170%;
            line-height: 30px;
            font-weight: 700;
            color: #de4d00;
            margin: 40px 0 10px;
        }

        h3 {
            font-weight: 700;
            font-size: 120%;
            margin-top: 20px;
        }
    </style>
</head>

<body style="margin: 0;">

    <div class="jumbotron" style="margin-bottom: 0px;">
        <div class="container">
            <table>
                <tbody>
                    <tr>
                        <td>
                            <h1>Hello, your <strong>ENA-upload Galaxy</strong> container is running!</h1><br>
                            <p> This container is optimized to clean human reads from raw reads and to upload them to
                                ENA,
                                this in only a few steps. The ENA uploading is based on the tool ena-upload-cli while
                                the
                                reads_cleaning tool is based on Metagen-FastQC workflow.
                            </p>
                        </td>
                        <td>
                            <div align="center">
                                <img src="./images/welcome_elixir_logo.png" style="margin:10px; height:100px;">
                            </div>

                        </td>
                    </tr>
                </tbody>
            </table>
        </div>
    </div>
    <div class="container mainbody">
        <h2 id="1-uploading-data-to-galaxy">1. Uploading data to Galaxy</h2>
        <p>ENA will accept only raw read files via programmatic submissions (this tool). Genome or transcriptome
            assemblies can be submitted to ENA using the the ENA website.
            ENA will accept <em>fastq.gz</em> (SE and PE), <em>bam</em> and <em>cram</em> formats.</p>

        <p>Raw data can be uploaded using uploaded using <code>Upload File</code>, found at the top of the
            <strong>Tools</strong> panel:</p>

        <figure class="docimage">
            <img src="./images/uploading_files_galaxy.png" alt="Uploading files In Galaxy" />
            <figcaption>
                Uploading data to Galaxy
            </figcaption>
        </figure>

        <p>There are different recommended options for uploading the read data:</p>
        <ul>
            <li>Load by “browsing” for a local file. Best for small datasets. Some servers will support load data that
                is 2 GB or larger. If you are having problems with this method, try FTP.</li>
            <li>Load using an HTTP URL or FTP URL.</li>
        </ul>

        <p>When the local files are selected or a url is given, click “start” to start the uploading to Galaxy.</p>

        <p>More information on data upload to Galaxy can be found in <a
                href="https://galaxyproject.org/support/loading-data/">Galaxy support</a>.</p>

        <p>Your data should appear on the right panel in green. You can rename, tag, preview edit or delete data objects
            from here.</p>
        <figure class="docimage">
            <img src="./images/history_galaxy.png" alt="history galaxy" />
            <figcaption>
                Files that are uploaded will show up in the history panel.
            </figcaption>
        </figure>

        <h2 id="2-filter-human-reads-out-of-the-raw-reads">2. Filter human reads out of the raw reads</h2>
        <p>In order to comply with Europe’s General Data Protection Regulation <a
                href="https://ec.europa.eu/info/law/law-topic/data-protection/eu-data-protection-rules_en">(GDPR)</a>,
            traces of human genetic information must be removed from the raw data before submitting it to ENA. A tool is
            included that filters out reads that map to the human genome using <a
                href="https://github.com/Finn-Lab/Metagen-FastQC">Metagen-FastQC</a>.</p>

        <p>Select the filtering tool from the <strong>Tools</strong> panel on the right.</p>

        <ul>
            <li>Select human h38 reference genome.</li>
            <li>Choose single or paired end</li>
            <li>Select the uploaded files</li>
            <li>Click on <em>Execute</em></li>
        </ul>

        <figure class="docimage">
            <img src="./images/read_cleaning_tool.png" alt="Read cleaning tool" />
            <figcaption>
                The interface of the read cleaning tool.
            </figcaption>
        </figure>

        <p>The tool will now process the raw reads to remove reads that map to the human genome. This can take a while.
            The resulting filtered data files are found on the right panel.</p>

        <h2 id="3-uploading-the-data-accompanied-with-the-metadata">3. Uploading the data accompanied with the metadata
        </h2>
        <p>Submission to ENA requires accompanying metadata that complies with the <a
                href="https://ena-docs.readthedocs.io/en/latest/submit/general-guide/metadata.html">ENA metadata
                model</a>.
            The tool offers two ways of entering metadata for submission: interactively or via a metadata template. All
            metadata fields must be completed for the submission to go through. Both ways allow you to make a test
            submission to the server of ENA. This enables you to check if everything is showing up as expected and the
            metadata is correct.</p>

        <figure class="docimage">
            <img src="./images/2_ways_ena_tool.png" alt="2ways of submitting metadata" />
            <figcaption>
                Two ways of submitting the metadata through the ENA-upload tool.
            </figcaption>
        </figure>

        <h3 id="interactive-metadata-upload">Interactive metadata upload</h3>
        <p>For a small number of submissions, metadata is best entered interactively by completing all the boxes.
            Metadata fields are nested according to ENA metadata model:</p>
        <ul>
            <li>A <strong>study (project)</strong> groups together data submitted to the archive and controls its
                release date. A study accession is typically used when citing data submitted to ENA.</li>
            <li>A <strong>sample</strong> contains information about the sequenced source material. Samples are
                associated with checklists, which define the fields used to annotate the samples. Samples are always
                associated with a taxon.</li>
            <li>An <strong>experiment</strong> contains information about a sequencing experiment including library and
                instrument details.</li>
            <li>A <strong>run</strong> is part of an experiment and refers to data files containing sequence reads.</li>
        </ul>

        <p>At the bottom, select the filtered data files associated with the metadata.</p>

        <p>At last, fill in the Affiliation center and click on execute.</p>

        <figure class="docimage">
            <img src="./images/execute_ena_upoad.png" alt="Affiliation center and execute button" />
            <figcaption>
                Affiliation field and exicute button.
            </figcaption>
        </figure>

        <h3 id="upload-with-the-metadata-template">Upload with the metadata template</h3>
        <p>For submission of a large number of files, it is recommended to use the spreadsheet template to upload the
            metadata. The template can be found <a
                href="https://docs.google.com/spreadsheets/d/14pHWToG7Hm5vLhToA6ex9K2oBFm0Wygao53ZsktzjPg/edit?usp=sharing">here</a>.
            The template is organized according to the ENA metadata model and contains one worksheet each for study,
            sample, experiment and run metadata. Use the alias column to interlink the experiments, studies, runs,
            samples and files with each other. The template can be downloaded, completed and uploaded using the Galaxy
            upload tool.</p>

        <p>Open the spreadsheet in your computer. <strong>All fields of the template must be complete</strong>. <a
                href="https://drive.google.com/file/d/1Z3LszV6IkgmcESsz2K7Mdv8pcnolUHT5/view?usp=sharing">Here</a> you
            can find an example of part of the metadata associated with a submission to ENA using this tool (accession
            number PRJEB40711).</p>

        <p>Export the tables to tsv files and upload the completed metadata template using Galaxy upload tool. Select
            the correct spreadsheets in the tool together with the correct data. If the data is uploaded correctly, they
            should appear in the “Select all datasets to upload” section.</p>

        <p>At last, fill in the Affiliation center and click on execute.</p>

        <figure class="docimage">
            <img src="./images/execute_ena_upoad.png" alt="Affiliation center and execute button" />
            <figcaption>
                Affiliation field and exicute button.
            </figcaption>
        </figure>

        <h2 id="4-check-for-a-valid-submission">4. Check for a valid submission</h2>

        <p>Visit <a href="https://www.ebi.ac.uk/ena/submit/webin">Webin online</a> to check on your submissions or <a
                href="https://wwwdev.ebi.ac.uk/ena/submit/webin">dev Webin</a> to check on test submissions. If
            everything looks fine, publish the data by changing the “Release Date” of the study to a day later than the
            current day. It can take several days for ENA to index the data and to let it appear in a correct manner.
            Covid-19 data will also be indexed by the <a href="https://www.covid19dataportal.org/">COVID-19 Data
                Portal</a></p>


    </div>

    <div class="footer">
        <div class="container padding-m">
            <p>
                <a target="_blank" class="reference" href="http://galaxyproject.org/">
                    Galaxy</a> is an open platform for supporting data intensive
                research. Galaxy is developed by <a target="_blank" class="reference"
                    href="http://wiki.galaxyproject.org/GalaxyTeam">The Galaxy Team</a>
                with the support of <a target="_blank" class="reference"
                    href="https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTORS.md">many contributors</a>.
                The Galaxy Docker project is supported by the University of Freiburg, part of de.NBI.
            </p>

            <p class="small">
                The <a target="_blank" class="reference" href="http://galaxyproject.org/">Galaxy Project</a>
                is supported in part by <a target="_blank" class="reference" href="http://www.genome.gov">NHGRI</a>,
                <a target="_blank" class="reference" href="http://www.nsf.gov">NSF</a>,
                <a target="_blank" class="reference" href="http://www.huck.psu.edu">The Huck Institutes of the Life
                    Sciences</a>,
                <a target="_blank" class="reference" href="http://www.ics.psu.edu">The Institute for CyberScience at
                    Penn
                    State</a>,
                and <a target="_blank" class="reference" href="http://www.jhu.edu/">Johns Hopkins University</a>.
            </p>
        </div>
    </div>
</body>

</html>